src/models/stores/class.ts

```diff
@@ -1,44 +1,60 @@
 import { types } from "mobx-state-tree";
 import { ClassInfo } from "../../lib/auth";
+import { values } from "lodash";
 
-export const ClassStudentModel = types
-  .model("ClassStudent", {
+export enum ClassUserType {
+  teacher = "teacher",
+  student = "student"
+}
+
+export const ClassUserModel = types
+  .model("ClassUser", {
+    type: types.enumeration<ClassUserType>("ClassUser", values(ClassUserType) as ClassUserType[]),
     id: types.string,
     firstName: types.string,
     lastName: types.string,
     fullName: types.string,
     initials: types.string,
+  })
+  .views((self) => {
+    return {
+      get displayName() {
+        return self.type === ClassUserType.student
+          ? self.fullName
+          : `Teacher ${self.lastName}`;
+      }
+    };
   });
 
 export const ClassModel = types
   .model("Class", {
     name: types.string,
     classHash: types.string,
-    students: types.array(ClassStudentModel),
+    users: types.array(ClassUserModel),
```

Comment:
Decided not to switch to a map here? Seems like it would be a better fit as we discussed earlier.

Reply:
Agreed it's a better fit, just didn't want to deal with the additional changes...but you convinced me! Updated to reflect this